platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MarkersFunctions.java

```diff
@@ -0,0 +1,262 @@
+package com.mapbox.mapboxsdk.maps;
+
+
+import android.graphics.RectF;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.util.LongSparseArray;
+
+import com.mapbox.mapboxsdk.annotations.Annotation;
+import com.mapbox.mapboxsdk.annotations.BaseMarkerOptions;
+import com.mapbox.mapboxsdk.annotations.BaseMarkerViewOptions;
+import com.mapbox.mapboxsdk.annotations.Icon;
+import com.mapbox.mapboxsdk.annotations.IconFactory;
+import com.mapbox.mapboxsdk.annotations.Marker;
+import com.mapbox.mapboxsdk.annotations.MarkerView;
+import com.mapbox.mapboxsdk.annotations.MarkerViewManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class MarkersFunctions implements Markers {
+
+  private final NativeMapView nativeMapView;
+  private final MapView mapView;
+  private final LongSparseArray<Annotation> annotations;
+  private final IconManager iconManager;
+  private final MarkerViewManager markerViewManager;
+
+  MarkersFunctions(NativeMapView nativeMapView, MapView mapView, LongSparseArray<Annotation> annotations, IconManager
```

Comment:
JavaDoc + Naming

In this case the plurality of Markers is also a bit weird here I think.